server/Arcadia.Assistant.Web/Controllers/CalendarEventsApprovalsController.cs

```diff
@@ -0,0 +1,166 @@
+        {
+            var employee = await this.GetEmployeeOrDefaultAsync(employeeId, token);
+            if (employee == null)
+            {
+                return this.NotFound();
+            }
+
+            var authorizationResult = await this.authorizationService.AuthorizeAsync(this.User, employee, new ReadCalendarEvents());
+            if (!authorizationResult.Succeeded)
+            {
+                return this.Forbid();
+            }
+
+            var requestedEvent = await this.GetCalendarEventOrDefaultAsync(employee.Calendar.CalendarActor, eventId, token);
+            if (requestedEvent == null)
+            {
+                return this.NotFound();
+            }
+
+            var response = await employee.Calendar.CalendarActor.Ask<GetCalendarEventApprovals.Response>(
+                new GetCalendarEventApprovals(requestedEvent),
+                this.timeoutSettings.Timeout,
+                token);
+
+            switch (response)
+            {
+                case GetCalendarEventApprovals.SuccessResponse result:
+                    var model = result.Approvals.Select(a => new CalendarEventApprovalModel(a));
+                    return this.Ok(model);
+
+                default:
+                    return this.StatusCode(StatusCodes.Status500InternalServerError);
+            }
+        }
+
+        [HttpPost]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [ProducesResponseType(StatusCodes.Status409Conflict)]
+        [ProducesResponseType(StatusCodes.Status202Accepted)]
+        public async Task<IActionResult> ApproveEvent(string employeeId, string eventId, [FromBody]CalendarEventApprovalModel model, CancellationToken token)
+        {
+            if (!this.ModelState.IsValid)
+            {
+                return this.BadRequest(this.ModelState);
+            }
+
+            var employee = await this.GetEmployeeOrDefaultAsync(employeeId, token);
+            if (employee == null)
+            {
+                return this.NotFound();
+            }
+
+            var approver = await this.GetEmployeeOrDefaultAsync(model.ApproverId, token);
+            if (approver == null)
+            {
+                return this.BadRequest(this.ModelState);
+            }
+
+            var authorizationResult = await this.authorizationService.AuthorizeAsync(this.User, approver, new EditPendingCalendarEvents());
+            if (!authorizationResult.Succeeded)
+            {
+                return this.Forbid();
+            }
+
+            var requestedEvent = await this.GetCalendarEventOrDefaultAsync(employee.Calendar.CalendarActor, eventId, token);
+            if (requestedEvent == null)
+            {
+                return this.NotFound();
+            }
+
+            if (!requestedEvent.IsPending)
```

Comment:
why is that? Can't I send approval for already approved request, for example? If I really want to :)

Reply:
anyway, you don't need to check that here because it throws an error from inside of actor (even if we want that check). And it throws it with a message

Reply:
Done.